src/User/Search/Gambit/GroupGambit.php

```diff
@@ -44,16 +45,14 @@ protected function conditions(AbstractSearch $search, array $matches, $negate)
             throw new LogicException('This gambit can only be applied on a UserSearch');
         }
 
-        $groupNames = $this->extractGroupNames($matches);
+        $groupIdentifiers = $this->extractGroupIdentifiers($matches);
 
-        // TODO: Use a JOIN instead (and don't forget to remove the findByName() method again)
-        $ids = [];
-        foreach ($groupNames as $name) {
-            $group = $this->groups->findByName($name);
-            if ($group && count($group->users)) {
-                $ids = array_merge($ids, $group->users->pluck('id')->all());
-            }
-        }
+        $ids = Group::whereIn('name_singular', $groupIdentifiers)
```

Comment:
I'd suggest taking the opportunity to rename this variable to `$userIds` to improve readability.